sdk/python/kfp/dsl/for_loop.py

```diff
@@ -156,15 +163,17 @@ def _make_name(self, code: str):
     @classmethod
     def from_pipeline_channel(
         cls,
-        channel: pipeline_channel.PipelineParameterChannel,
+        channel: pipeline_channel.PipelineChannel,
+        is_artifact_list: bool = False,
     ) -> 'LoopArgument':
-        """Creates a LoopArgument object from a PipelineParameterChannel
-        object."""
+        """Creates a LoopArgument object from a PipelineChannel object."""
         return LoopArgument(
             items=channel,
             name_override=channel.name + '-' + cls.LOOP_ITEM_NAME_BASE,
             task_name=channel.task_name,
             channel_type=_get_loop_item_type(channel.channel_type) or 'String',
+            is_artifact_list=is_artifact_list,
+            value=None,
```

Comment:
I commented out lines 176 and 193 and the test still passed. Do we need an additional test case for this?